ports/stm32/mboot/mboot.h

```diff
@@ -132,4 +149,20 @@ static inline void mboot_entry_init(uint32_t *initial_r0) {
     #endif
 }
 
+#if defined(MBOOT_BOARD_GET_RESET_MODE)
+static inline int mboot_get_reset_mode(void) {
+    return MBOOT_BOARD_GET_RESET_MODE();
+}
+#else
+int mboot_get_reset_mode(void);
+#endif
+
+#if defined(MBOOT_BOARD_STATE_CHANGE)
+static inline void mboot_state_change(mboot_state_t state, uint32_t arg) {
```

Comment:
If `mboot_state_change()` is called in more than one file, this will result in multiple copies of the function (unless linker optimization is enabled, which it usually isn't). Perhaps it would be better to leave out the `#if` here and move this to the `.c` file instead of the header?

Reply:
The idea is that a board would do something like:
```c
#define MBOOT_BOARD_STATE_CHANGE my_board_state_change
```
Then it's efficient, and effectively just renaming the function.

Reply:
I guess it is fine if it is used that way. It just seems like GCC doesn't ever actually inline `inline` functions.

Reply:
That isn't what disassembly tells me (assuming my memory is correct, which is not always a good assumption) - but that is probably a discussion for a later time.